java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S3329.html

```diff
@@ -4,54 +4,40 @@
 <p>To generate Initialization Vectors, NIST recommends to use a secure random number generator.</p>
 <h2>Noncompliant Code Example</h2>
 <pre>
-public class MyCbcClass {
+public void encrypt(String key, String plainText) {
+    byte[] bytesIV = "7cVgr5cbdCZVw5WY".getBytes("UTF-8"); // Static initialization vector
 
-  public String applyCBC(String strKey, String plainText) {
-    byte[] bytesIV = "7cVgr5cbdCZVw5WY".getBytes("UTF-8");
-
-    /* KEY + IV setting */
     IvParameterSpec iv = new IvParameterSpec(bytesIV);
     SecretKeySpec skeySpec = new SecretKeySpec(strKey.getBytes("UTF-8"), "AES");
 
-    /* Ciphering */
-    Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
+    Cipher cipher = Cipher.getInstance("AES/CBC/NoPadding");
     cipher.init(Cipher.ENCRYPT_MODE, skeySpec, iv);  // Noncompliant: the IV is hard coded and thus not generated with a secure random generator
-    byte[] encryptedBytes = cipher.doFinal(plainText.getBytes("UTF-8"));
-    return DatatypeConverter.printBase64Binary(bytesIV)
-            + ";" + DatatypeConverter.printBase64Binary(encryptedBytes);
-  }
 }
 </pre>
 <h2>Compliant Solution</h2>
 <pre>
-public class MyCbcClass {
-
-  SecureRandom random = new SecureRandom();
-
-  public String applyCBC(String strKey, String plainText) {
+public void encrypt(String key, String plainText) {
+    SecureRandom random = new SecureRandom();
     byte[] bytesIV = new byte[16];
-    random.nextBytes(bytesIV);
+    random.nextBytes(bytesIV); // Random initialization vector
 
-    /* KEY + IV setting */
     IvParameterSpec iv = new IvParameterSpec(bytesIV);
     SecretKeySpec skeySpec = new SecretKeySpec(strKey.getBytes("UTF-8"), "AES");
 
-    /* Ciphering */
-    Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
-    cipher.init(Cipher.ENCRYPT_MODE, skeySpec, iv); // Compliant
-    byte[] encryptedBytes = cipher.doFinal(plainText.getBytes("UTF-8"));
-    return DatatypeConverter.printBase64Binary(bytesIV)
-            + ";" + DatatypeConverter.printBase64Binary(encryptedBytes);
-  }
+    Cipher cipher = Cipher.getInstance("AES/CBC/NoPadding");
+    cipher.init(Cipher.ENCRYPT_MODE, skeySpec, iv);
 }
```

Comment:
The new code does not compile because `strKey` should be `key`.
```java
  public void encrypt(String key, String plainText) throws GeneralSecurityException {
    SecureRandom random = new SecureRandom();
    byte[] bytesIV = new byte[16];
    random.nextBytes(bytesIV); // Random initialization vector

    IvParameterSpec iv = new IvParameterSpec(bytesIV);
    SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes(StandardCharsets.UTF_8), "AES");

    Cipher cipher = Cipher.getInstance("AES/CBC/NoPadding");
    cipher.init(Cipher.ENCRYPT_MODE, skeySpec, iv);
  }
```